doc/adr/0007-marconi-indexer.rst

```diff
@@ -0,0 +1,118 @@
+.. _marconi_indexer:
+
+ADR X: Marconi indexers
+=======================
+
+Date: 2022-08-05
+
+Authors
+-------
+
+Ometita Radu <radu.ometita@iohk.io>
+
+Status
+------
+
+Draft
+
+Context
+-------
+
+Off-chain code needs access to indexed portions of the blockchain. Currently, we have a working solution in the form of the chain-index and PAB (which both index information). The big problem with the current solution is its lack of reuse (or modularity) capability.
+
+We attempt to fix that problem with Marconi where we currently have a generic indexer that stores volatile information in memory and blocks that have been fully committed (are old enough to guarantee that they will not be rolled back). The K blockchain parameter represents the number of slots after which a block can no longer be rolled back.
+
+We currently need to keep K blocks in memory to be able to perform rollbacks. However, the K parameter can be adjusted for indexers which land us in the unfortunate position of saying that there may be data corruption in the case where the number of rollbacked blocks is larger than the number of blocks stored in memory. While this is both detectable and unlikely to happen we think that our current solution can prevent it without any significant drawbacks.
+
+Decision
+--------
+
+Initially, we needed a generic way of indexing information where we can control the amount of memory the indexer uses. The initial solution was to store the volatile blocks in memory and persist them on disk whenever they become older than the K parameter.
```

Comment:
Which initial solution? This ADR is actually about documenting this "initial" solution, no?

Reply:
Initial solution -> PAB/chain-index